homeassistant/components/asuswrt/bridge.py

```diff
@@ -0,0 +1,398 @@
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from .const import (
+    CONF_DNSMASQ,
+    CONF_INTERFACE,
+    CONF_REQUIRE_IP,
+    CONF_SSH_KEY,
+    DEFAULT_DNSMASQ,
+    DEFAULT_INTERFACE,
+    PROTOCOL_HTTP,
+    PROTOCOL_HTTPS,
+    PROTOCOL_TELNET,
+    SENSORS_BYTES,
+    SENSORS_LOAD_AVG,
+    SENSORS_RATES,
+    SENSORS_TEMPERATURES,
+)
+
+SENSORS_TYPE_BYTES = "sensors_bytes"
+SENSORS_TYPE_COUNT = "sensors_count"
+SENSORS_TYPE_LOAD_AVG = "sensors_load_avg"
+SENSORS_TYPE_RATES = "sensors_rates"
+SENSORS_TYPE_TEMPERATURES = "sensors_temperatures"
+
+WrtDevice = namedtuple("WrtDevice", ["ip", "name", "connected_to"])
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_dict(keys: list, values: list) -> dict[str, Any]:
+    """Create a dict from a list of keys and values."""
+    ret_dict: dict[str, Any] = dict.fromkeys(keys)
+
+    for index, key in enumerate(ret_dict):
+        ret_dict[key] = values[index]
+
+    return ret_dict
+
+
+class AsusWrtBridge(ABC):
+    """The Base Bridge abstract class."""
+
+    @staticmethod
+    def get_bridge(
+        hass: HomeAssistant, conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> AsusWrtBridge:
+        """Get Bridge instance."""
+        protocol = conf[CONF_PROTOCOL]
+        if protocol in [PROTOCOL_HTTP, PROTOCOL_HTTPS]:
+            session = async_get_clientsession(hass)
+            return AsusWrtHttpBridge(conf, session)
+        return AsusWrtLegacyBridge(conf, options)
+
+    def __init__(self) -> None:
+        """Initialize Bridge."""
+        self._firmware: str | None = None
+        self._label_mac: str | None = None
+        self._model: str | None = None
+
+    @property
+    def firmware(self) -> str | None:
+        """Return firmware information."""
+        return self._firmware or None
+
+    @property
+    def label_mac(self) -> str | None:
+        """Return label mac information."""
+        return self._label_mac or None
+
+    @property
+    def model(self) -> str | None:
+        """Return model information."""
+        return self._model or None
+
+    @property
+    @abstractmethod
+    def is_connected(self) -> bool:
+        """Get connected status."""
+
+    @abstractmethod
+    async def async_connect(self) -> None:
+        """Connect to the device."""
+
+    @abstractmethod
+    async def async_disconnect(self) -> None:
+        """Disconnect to the device."""
+
+    @abstractmethod
+    async def async_get_connected_devices(self) -> dict[str, WrtDevice]:
+        """Get list of connected devices."""
+
+    @abstractmethod
+    async def async_get_available_sensors(self) -> dict[str, dict[str, Any]]:
+        """Return a dictionary of available sensors for this bridge."""
+
+
+class AsusWrtLegacyBridge(AsusWrtBridge):
```

Comment:
I think this should be removed, there is no real point in dividing into 2 versions?

Reply:
And when doing that also remove me as code-owner would be good. (moved on from hass)

Reply:
My idea was to maintain both considering the variety of routers in circulation. Can we wait for other opinions?

Reply:
I honestly don't understand this comment and why you put it here in this PR which needs to eventually be reviewed by the HA teams. You are free to test and use any custom integration and I believe no one has forced you to do so, but complaining about a problem demanding a solution without giving details about the problems doesn't work at all. I don't think you are paid to test and I am not paid to develop code, my only purpose is to help improve this solution by providing my programming skills for my enjoyment. So if you would like support on custom integration, open an issue in the custom integration repository and provide details on what isn't working, or if you prefer stick with what works and live your life.

Reply:
I have just finished "evaluate" Asus_Router-Custom integration, and i fall back to ASUSWRT, Would be great if You could bring this Integration upto-date, and maybe as you mentioned in "Community-Topic" ADD a few more sensors.
What first come to my mind, was (beside External IP-Sensor) also Router-uptime.sensor

HA " UPnP/IGD " Core-Integration, include these 2 sensors, but from a usage-perspective would it be great if we don't have to use 2 different Core-Integrations, to get info's from the Router 
Personally im not interested in DL/UL-Total or Speed etc., But essential info and "Tracking" initially, And using only 1 integration for this :)